Support IAM roles on BigQuery datasets

rolandkool opened the pull request:
To fix https://github.com/terraform-providers/terraform-provider-google/issues/3206

danawillow commented:
Holding off review until the questions in #3206 are resolved.

hanfi commented:
hi @rolandkool 
i updated the documentation on your fork :+1 

https://github.com/rolandkool/terraform-provider-google/pull/1/files

rolandkool commented:
> hi @rolandkool
> i updated the documentation on your fork :+1
> 
> https://github.com/rolandkool/terraform-provider-google/pull/1/files

@hanfi, thanks! I've fixed one small typo and merged it. Hope this can make it into the next release.
